dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -3601,68 +3601,55 @@ ArithmeticColumn* buildArithmeticColumn(
 
     // @bug5715. Use InfiniDB adjusted coltype for result type.
     // decimal arithmetic operation gives double result when the session variable is set.
-    CalpontSystemCatalog::ColType mysql_type = colType_MysqlToIDB(item);
+    CalpontSystemCatalog::ColType mysqlType = colType_MysqlToIDB(item);
 
-    if (mysql_type.colDataType == CalpontSystemCatalog::DECIMAL ||
-        mysql_type.colDataType == CalpontSystemCatalog::UDECIMAL)
+    CalpontSystemCatalog::ColType leftColType = pt->left()->data()->resultType();
```

Comment:
resultType() returns a const IIRC so leftColType can be a const symbol.

Reply:
Done.

Reply:
Also replaced the OR code with an inline isDecimalOperands().

Reply:
Thx a lot.